### PR TITLE
Replaced ClientError and ServerError with WriteError in ResponseWriter

### LINKED ISSUE #127: Proposal: merge `ClientError` and `ServerError`
I think w.Error(Status, optionalResponse) could be enough, the response writer has enough information to decide what to do when given a status that is >400.

### PR DESCRIPTION
**Pull request by mattiasgrenfeldt**
Fixes #127

### CODE REVIEW

**Review by kele — APPROVED** [review]

File: safehttp/response_writer.go [hunk 1]
```diff
@@ -89,26 +89,10 @@ func (w *ResponseWriter) NoContent() Result {
 	return Result{}
 }
 
-// ClientError TODO
-func (w *ResponseWriter) ClientError(code StatusCode) Result {
+// WriteError writes an error response (4xx, 5xx) according to the provided status
```
Comment (mihalimara22) [inline safehttp/response_writer.go:92]: Technically, the user could provide any status code so maybe make it more explicit that a valid status code should be between 400 and 599.

**Review by mihalimara22 — APPROVED** [review]

**Comment by mattiasgrenfeldt** [comment]
Ready to merge! @kele @empijei

**Comment by mattiasgrenfeldt** [comment]
I'll rebase this now

**Comment by mattiasgrenfeldt** [comment]
Ready again! @kele

**Pull request merged**